### PR TITLE
Swagger adjustment: Supported pool id param is hex or bech32

### PR DESCRIPTION
**Pull request by piotr-iohk**
# Issue Number

https://github.com/input-output-hk/cardano-wallet/issues/2023
https://github.com/input-output-hk/cardano-wallet/issues/2025

# Overview

- 01d5d509e65fa9ba18f2be91554953ca2ccdb15d
  Supported pool id param is hex or bech32

# Comments

**Before:**

Screenshot from 2020-08-31 13-42-59

**After:**

Screenshot from 2020-08-31 16-16-37

### CODE REVIEW

File: specifications/api/swagger.yaml [hunk 1]
```diff
@@ -1404,9 +1404,9 @@ x-parametersStakePoolId: &parametersStakePoolId
   required: true
   schema:
     type: string
-    format: hex
+    format: hex|bech32
     maxLength: 56
-    minLength: 64
+    minLength: 56
```
Comment (piotr-iohk) [inline specifications/api/swagger.yaml]: Empirically I see that pool ids (no matter if in hex or bech32 format) have consistently 56 char length :man_shrugging:

Comment (hasufell) [reply specifications/api/swagger.yaml]: @KtorZ is that something we can guarantee? I'm not sure how deterministic bech32 is here.

Comment (KtorZ) [reply specifications/api/swagger.yaml]: I don't think this is right indeed, so we could leave it as a comment in the description instead.

Also, pool ids can be 64 chars ..... with jormungandr :s

Comment (piotr-iohk) [reply specifications/api/swagger.yaml]: I have added a comment. Pls let me know it that's fine.

File: specifications/api/swagger.yaml [hunk 2]
```diff
@@ -2226,6 +2224,9 @@ paths:
         <p align="right">status: <strong>stable</strong></p>
 
         Delegate all (current and future) addresses from the given wallet to the given stake pool.
+
+        <strong>Note:</strong> Stake pool id lenght can be undeterministic when encoded as bech32.
```
Comment (hasufell) [inline specifications/api/swagger.yaml]: `length`

**Review by hasufell — APPROVED** [review]

File: specifications/api/swagger.yaml (hunk 2 above)
Comment (piotr-iohk) [reply specifications/api/swagger.yaml]: :facepalm:

File: specifications/api/swagger.yaml [hunk 3]
```diff
@@ -2226,6 +2224,9 @@ paths:
         <p align="right">status: <strong>stable</strong></p>
 
         Delegate all (current and future) addresses from the given wallet to the given stake pool.
+
+        <strong>Note:</strong> Stake pool id length can be undeterministic when encoded as bech32.
```
Comment (KtorZ) [inline specifications/api/swagger.yaml]: can be --> is ?

**Review by KtorZ — APPROVED** [review]

File: specifications/api/swagger.yaml (hunk 3 above)
Comment (jonathanknowles) [reply specifications/api/swagger.yaml]: I'm not sure that "undeterministic" (which should probably be "nondeterministic") is the best way to word this.

Readers **_might_** infer that the bech32 encoding is nondeterministic in its length. Of course this is not true, as Bech32 encoding is completely deterministic.

Perhaps we could instead say something like:

```
Bech32-encoded stake pool identifiers can vary in length.
```

And perhaps then give some examples to illustrate the different lengths that are possible.

@piotr-iohk What do you think?

**Review by jonathanknowles — COMMENTED** [review]
@piotr-iohk 

I'm not sure that "undeterministic" (which should probably read "nondeterministic") is the best way to word this.

Unfamiliar readers _might_ infer that the bech32 encoding is nondeterministic in its length. (Of course as we know this isn't true, as Bech32 encoding is completely deterministic.)

Perhaps we could instead say something like:

```
Bech32-encoded stake pool identifiers can vary in length.
```

And perhaps then give some examples to illustrate the different lengths that are possible.

What do you think?

**Pull request merged**